nntrainer/graph/network_graph.cpp

```diff
@@ -440,25 +459,31 @@ int NetworkGraph::isCompilable() {
     return ML_ERROR_INVALID_PARAMETER;
   }
 
-  Layer &l = *adj[0].front().layer;
-
-  /** Dimension of first layer must be known */
-  // TODO: move this to the input layer and not first layer
-  if (l.getInputDimension().size() == 0) {
-    ml_loge("InputDimension of first layer is not set");
-    return ML_ERROR_INVALID_PARAMETER;
-  }
+  return ML_ERROR_NONE;
+}
 
+int NetworkGraph::checkCompiledGraph() {
+  auto &l = Sorted[0].layer;
```

Comment:
Check if Sorted.empty()?

Reply:
this is called internally in `Compile`, and sorted is guaranteed to be non-empty by `topologicalSort`.

Reply:
Cool